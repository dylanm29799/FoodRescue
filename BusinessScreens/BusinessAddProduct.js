import React, { useState, useEffect } from "react";
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Button,
  Image,
  ColorPropType,
} from "react-native";
import Colour from "../constants/Colour";
import { scale } from "../components/ResponsiveText";
import * as ImagePicker from "expo-image-picker";
import CheckBox from "@react-native-community/checkbox";
import foodcountdown from "../models/foodcountdown";
import * as firebase from "firebase";
import "firebase/firestore";
import ButtonCustom from "../constants/ButtonCustom";
import "firebase/storage";
import { Picker } from "@react-native-picker/picker";
import { set } from "react-native-reanimated";
import NormalProduct from "../components/NormalProduct";
import FoodCountdown from "../components/FoodCountdown";

const BusinessAddProduct = (props) => {
  const [npview, setNpview] = useState("picked");
  const [nptext, setNptext] = useState("pickedText");
  const [fcview, setFcview] = useState("notPicked");
  const [fctext, setFctext] = useState("notPickedText");
  const [borderCorner, setBorderCorner] = useState({ borderTopLeftRadius: 30 });
  const [borderCornerFood, setBorderCornerFood] = useState({
    borderTopRightRadius: 30,
  });
  const [renderedPage, setRenderedPage] = useState("Normal");

  onFoodCountdown = () => {
    setFctext("pickedText");
    setFcview("picked");
    setNptext("notPickedText");
    setNpview("notPicked");
    setBorderCorner({ borderTopLeftRadius: 30 });
    console.log("FoodCountdown Pressed");

    setRenderedPage("FoodCountdown");
    console.log(renderedPage);
  };

  onNormalProduct = () => {
    setFctext("notPickedText");
    setFcview("notPicked");
    setNptext("pickedText");
    setNpview("picked");
    setBorderCornerFood({ borderTopRightRadius: 30 });

    console.log("Normal Pressed");

    setRenderedPage("Normal");
    console.log(renderedPage);
  };

  const Render = () => {
    if (renderedPage == "Normal") {
      return <NormalProduct />;
    } else {
      return <FoodCountdown />;
    }
  };
  return (
    <View style={styles.container}>
      <View style={styles.category}>
        <TouchableOpacity
          //prettier-ignore
          style={[styles[npview],  borderCorner ]}
          onPress={onNormalProduct}
        >
          <Text style={styles[nptext]}>Normal Product</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles[fcview], borderCornerFood]}
          onPress={onFoodCountdown}
        >
          <Text style={styles[fctext]}>Food Countdown</Text>
        </TouchableOpacity>
      </View>
      <Render />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  content: {
    borderTopWidth: 0,
    borderWidth: 2,
    borderColor: Colour.primaryColour,

    height: scale(500),
    width: scale(250),
    justifyContent: "center",
    alignItems: "center",
  },
  input: {
    borderBottomWidth: 2,
    borderBottomColor: Colour.primaryColour,
    width: "70%",
    paddingVertical: scale(3),
    paddingHorizontal: scale(10),
    textAlign: "center",
  },

  text: {
    marginTop: scale(10),
    fontSize: scale(13),
    textAlign: "center",
    fontFamily: "OpenSans",
  },

  touchable: {
    height: scale(30),
    width: scale(70),
    backgroundColor: Colour.primaryColour,
    justifyContent: "center",

    marginHorizontal: scale(10),
  },
  touchableText: {
    color: "#fff",

    textAlign: "center",
    fontSize: scale(12),
  },
  picked: {
    backgroundColor: Colour.primaryColour,
    flex: 1,
    justifyContent: "center",
    elevation: 100,
  },

  notPicked: {
    backgroundColor: "#fff",
    flex: 1,
    justifyContent: "center",

    borderWidth: 2,
    borderColor: Colour.primaryColour,
  },

  pickedText: {
    color: "#fff",
    textAlign: "center",
    fontSize: scale(12),
    fontFamily: "OpenSans",
  },
  notPickedText: {
    color: Colour.primaryColour,
    textAlign: "center",
    fontSize: scale(12),
    fontFamily: "OpenSans",
  },

  category: {
    justifyContent: "center",
    flexDirection: "row",
    width: "90%",
    height: scale(50),
  },
});

export default BusinessAddProduct;
