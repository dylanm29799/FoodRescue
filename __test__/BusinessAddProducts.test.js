/*
 *
 * ClassName: // __tests__/BusinessAddProduct.test.js
 *
 * Date: 28/04/2021
 *
 *
 * @author: Dylan Murphy, X17506166
 *
 * @reference : https://docs.expo.io/guides/testing-with-jest/
 * @reference : https://medium.com/react-native-training/learning-to-test-react-native-with-jest-part-1-f782c4e30101
 *
 */
import React from "react";
import BusinessAddProduct from "../BusinessScreens/BusinessAddProduct";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

jest.mock("react-navigation", () => ({
  withNavigation: (Component) => (props) => (
    <Component navigation={{ navigate: jest.fn() }} {...props} />
  ),
}));

test("renders correctly", () => {
  const tree = renderer.create(<BusinessAddProduct />).toJSON();
  expect(tree).toMatchSnapshot();
});
