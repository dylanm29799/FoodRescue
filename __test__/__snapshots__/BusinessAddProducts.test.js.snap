// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 107.14285714285714,
        "justifyContent": "center",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#c44536",
          "borderTopLeftRadius": 30,
          "elevation": 100,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "textAlign": "center",
          }
        }
      >
        Normal Product
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#c44536",
          "borderTopRightRadius": 30,
          "borderWidth": 2,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#c44536",
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "textAlign": "center",
          }
        }
      >
        Food Countdown
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#c44536",
        "borderTopWidth": 0,
        "borderWidth": 2,
        "height": "80%",
        "justifyContent": "center",
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingBottom": 21.428571428571427,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        Product Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Pizza Slice"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#c44536",
            "borderRadius": 5,
            "borderWidth": 2,
            "fontFamily": "MonM",
            "fontSize": 32.14285714285714,
            "justifyContent": "center",
            "padding": 6.428571428571429,
            "paddingLeft": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        Quantity Available
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="8"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#c44536",
            "borderRadius": 5,
            "borderWidth": 2,
            "fontFamily": "MonM",
            "fontSize": 32.14285714285714,
            "justifyContent": "center",
            "padding": 6.428571428571429,
            "paddingLeft": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        Initial Price before discount
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="4.00"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#c44536",
            "borderRadius": 5,
            "borderWidth": 2,
            "fontFamily": "MonM",
            "fontSize": 32.14285714285714,
            "justifyContent": "center",
            "padding": 6.428571428571429,
            "paddingLeft": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 25.714285714285715,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        New Price for Food Rescue
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="2.40"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#c44536",
            "borderRadius": 5,
            "borderWidth": 2,
            "fontFamily": "MonM",
            "fontSize": 32.14285714285714,
            "justifyContent": "center",
            "padding": 6.428571428571429,
            "paddingLeft": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 25.714285714285715,
          "paddingTop": 21.428571428571427,
        }
      }
    >
      Time Available
    </Text>
    <View
      style={
        Object {
          "height": 50,
          "width": "50%",
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "1 Hour",
              "testID": undefined,
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "3",
            },
            Object {
              "label": "4 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "4",
            },
            Object {
              "label": "5 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "5",
            },
            Object {
              "label": "6 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "6",
            },
            Object {
              "label": "7 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "7",
            },
            Object {
              "label": "8 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "8",
            },
            Object {
              "label": "9 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "9",
            },
            Object {
              "label": "10 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "10",
            },
            Object {
              "label": "11 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "11",
            },
            Object {
              "label": "12 Hours",
              "testID": undefined,
              "textColor": undefined,
              "value": "12",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 16,
          "paddingVertical": 10,
        }
      }
    >
      Pick an image
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#c44536",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#EBF38B",
            "fontFamily": "MonM",
            "fontSize": 38.57142857142857,
            "textAlign": "center",
            "width": 321.42857142857144,
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
